prime-router/src/main/kotlin/ElementErrors.kt

```diff
@@ -23,39 +23,50 @@ data class MissingFieldMessage(
 data class InvalidDateMessage(
     override val type: ResponseMsgType = ResponseMsgType.INVALID_DATE,
     val formattedValue: String = "",
-    val fieldMapping: String = ""
+    val fieldMapping: String = "",
+    val format: String? = ""
 ) : ResponseMessage {
     override fun detailMsg(): String {
-        return "Invalid date: '$formattedValue' for element $fieldMapping"
+        if (format !== null) {
+            return "Invalid date: '$formattedValue' for element $fieldMapping. Reformat to $format."
```

Comment:
Refactor this, so you are not pasting the same message twice.  E.g.
val msg = "Invalid date: '$formattedValue' for element $fieldMapping."
if (format != null) msg += " Reformat to $format."

Futhermore, you use this same "Reformat to $format." in other places, so refactor to reuse it.